sceptre/config.py

```diff
@@ -178,7 +178,8 @@ def get_config(path):
                         environment_variable=os.environ,
                         var=user_variables,
                         environment_path=self.environment_path.split("/"),
-                        environment_config=environment_config
+                        environment_config=environment_config,
+                        stack_config=self
```

Comment:
Whats intended to be made available here? - At this point dictionary values of stack_config is still being read, hence won't be and will be confusing. If its just the `stack_config.name` that we want, then maybe we should look at including it in the environment_path?

Reply:
The template name isn't part of the environment_path though, it's an attribute of the template/stack.

Currently it contains 
name < what i want
sceptre_dir < not sure why you would need that
environment_path < already exposed
dependencies < could be of interest to hooks

I was just reluctant to add a specific variable for my needs when something generic could work.

Reply:
Agree with above - also take back adding to the environment path as would change the API for existing config. Dependencies still wouldn't available at this point - as config hasn't been "read" in. Also in this context `name` is actually just the base_file_name. So could just add `basename=self.name` -> {{ basename }}